code/modules/projectiles/guns/specialist.dm

```diff
@@ -167,6 +218,54 @@
 
 	return blocked
 
+// Snipers may enable or disable their laser tracker at will.
+/datum/action/item_action/specialist/toggle_laser
+	ability_primacy = SPEC_PRIMARY_ACTION_2
+
+/datum/action/item_action/specialist/toggle_laser/New(var/mob/living/user, var/obj/item/holder)
+	..()
+	name = "Toggle Tracker Laser"
+	button.name = name
+	button.overlays.Cut()
+	var/image/IMG = image('icons/mob/hud/actions.dmi', button, "sniper_toggle_laser_on")
+	button.overlays += IMG
+	update_button_icon()
+
+/datum/action/item_action/specialist/toggle_laser/update_button_icon()
+	var/obj/item/weapon/gun/rifle/sniper/sniper_rifle = holder_item
+
+	var/icon = 'icons/mob/hud/actions.dmi'
+	var/icon_state = "sniper_toggle_laser_[sniper_rifle.enable_aimed_shot_laser ? "on" : "off"]"
+
+	button.overlays.Cut()
+	var/image/IMG = image(icon, button, icon_state)
+	button.overlays += IMG
+
+/datum/action/item_action/specialist/toggle_laser/can_use_action()
+	var/mob/living/carbon/human/human = owner
+	var/obj/item/weapon/gun/rifle/sniper/sniper_rifle = holder_item
+
+	if(!istype(human) || human.is_mob_incapacitated())
+		return FALSE
+
+	if(sniper_rifle != human.r_hand && sniper_rifle != human.l_hand)
```

Comment:
there is a proper proc for this

Reply:
and you don't say which one it is okay man. okay .

Reply:
dobne